sdk/demos/intel/fluid/demo.js

```diff
@@ -0,0 +1,693 @@
+
+  let radioSimple = createElement('input',
+    {'type': 'radio',
+     'name': 'mode',
+     'id': 'radio1',
+     'value': 'simple',
+     'onchange': 'changeMode(this.value)',
+     'checked': 'checked'});
+  let lableSimple = createElement('label', {}, 'Simple N^2');
+  let dRadioSimple = createElement('div', {}, [radioSimple, lableSimple]);
+
+  let radioShared = createElement('input',
+    {'type': 'radio',
+     'name': 'mode',
+     'id': 'radio1',
+     'onchange': 'changeMode(this.value)',
+     'value': 'shared'});
+  let lableShared = createElement('label', {}, 'Shared Memory N^2');
+  let dRadioShared = createElement('div', {}, [radioShared, lableShared]);
+
+  let radioGrid = createElement('input',
+    {'type': 'radio',
+     'name': 'mode',
+     'id': 'radio1',
+     'onchange': 'changeMode(this.value)',
+     'value': 'grid'});
+  let lableGrid = createElement('label', {}, 'Grid + Sort');
+  let dRadioGrid = createElement('div', {}, [radioGrid, lableGrid]);
+
+  container.appendChild(dButton);
+  container.appendChild(dSelectP);
+  container.appendChild(dSelectG);
+  container.appendChild(dRadioSimple);
+  container.appendChild(dRadioShared);
+  container.appendChild(dRadioGrid);
+}
+
+function changeMode(o) {
+  gSimMode = o;
+  gFPSCounter.reset();
+  gFpsElem.innerHTML = 'calculating frames per second...';
+}
+
+function changeGravity(g) {
+  let data1 = new ArrayBuffer(8);
+  let view1 = new Float32Array(data1);
+  switch (g) {
+    case 'up':
+      gGravityX = view1[0] = 0.0;
+      gGravityY = view1[1] = 0.5;
+      break;
+    case 'down':
+      gGravityX = view1[0] = 0.0;
+      gGravityY = view1[1] = -0.5;
+      break;
+    case 'left':
+      gGravityX = view1[0] = -0.5;
+      gGravityY = view1[1] = 0.0;
+      break;
+    case 'right':
+      gGravityX = view1[0] = 0.5;
+      gGravityY = view1[1] = 0.0;
+      break;
+  }
+
+  gl.bindBuffer(gl.UNIFORM_BUFFER, gUpdateParams);
+  gl.bufferSubData(gl.UNIFORM_BUFFER, 12*4, view1, 0);
+  gl.bindBufferBase(gl.UNIFORM_BUFFER, 0, gUpdateParams);
+}
+
+function resetParticles() {
+  destroyBuffers();
+  initBuffers();
+  changeMode(gSimMode);
+  initRender();
+}
+
+function restart(num) {
+  gNumParticles = num * 1024;
+  gDispatchNum = Math.ceil(gNumParticles / gThreads);
+  destroyBuffers();
+  initBuffers();
+  changeMode(gSimMode);
+  initRender();
+}
+
+function gpuSort() {
+  let NUM_ELEMENTS = gNumParticles;
+  let MATRIX_WIDTH = BITONIC_BLOCK_SIZE;
+  let MATRIX_HEIGHT = NUM_ELEMENTS / BITONIC_BLOCK_SIZE;
+
+  // Sort the data
+  // First sort the rows for the levels <= to the block size
+  for ( let level = 2; level <= BITONIC_BLOCK_SIZE; level <<= 1 ) {
+    let sortCB = new Int32Array(4);
+    sortCB[0] = level;
+    sortCB[1] = level;
+    sortCB[2] = MATRIX_HEIGHT;
+    sortCB[3] = MATRIX_WIDTH;
+    let updateSortCB = gl.createBuffer();
+    gl.bindBuffer(gl.UNIFORM_BUFFER, updateSortCB);
+    gl.bufferData(gl.UNIFORM_BUFFER, sortCB, gl.STATIC_DRAW);
```

Comment:
Maybe above three lines can be put into initBuffers. Here you only need `gl.bufferSubData` to update the content. And don't forget to destroy them when reset.

Reply:
Done.